lib/identity/client.js

```diff
@@ -766,6 +766,51 @@ class Client extends PartialClient {
     )
     return await validateResponseAsJSON(response)
   }
+  /**
+   * removeSingleSecret
+   * @param {String} secretID
+   * @param {String} version
+   * @param {String} secretName
+   * @param {String} secretType
+   * @return {Boolean} returns true if the secret was successfully removed
+   */
+  async removeSingleSecret(secretID, version, secretName, secretType) {
+    const sharedList = await this.getSecretSharedList(secretName, secretType)
+    await Promise.all(
+      sharedList.map(async users => {
+        let identity = await this.searchIdentityByUsername(users.username)
+        const groupName = `tozny.secret.${this.config.realmName}.${this.storage.config.clientId}.${identity.client_id}.${secretName}.${secretType}`
+        const groupInfo = await this.storage.groupInfo(groupName)
+        const sharedRecords = await this.storage.listRecordsSharedWithGroup(
+          groupInfo.groupID
+        )
+        if (sharedRecords[0].length < 2) {
+          let recordType = `tozny.secret.${SECRET_UUID}.${secretType}.${secretName}`
+          await this.storage.revokeRecordWithGroup(
+            groupInfo.groupID,
+            recordType
+          )
+          await this.storage.deleteGroup(groupInfo.groupID)
+        }
+      })
+    )
+    const groupName = `tozny.secret.${this.config.realmName}.${
```

Comment:
almost all of this code is a duplicate of what's in lines 782, which is a clear sign it should be extracted out into a re-usable method